metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/AbstractMetisPlugin.java

```diff
@@ -18,8 +18,8 @@
 @JsonSubTypes({
     @JsonSubTypes.Type(value=OaipmhHarvestPlugin.class, name="OAIPMH_HARVEST"),
     @JsonSubTypes.Type(value=HTTPHarvestPlugin.class, name="HTTP_HARVEST"),
-    @JsonSubTypes.Type(value=DereferencePlugin.class, name="DEREFERENCE"),
-    @JsonSubTypes.Type(value=VoidMetisPlugin.class, name="VOID")
+    @JsonSubTypes.Type(value=EnrichmentPlugin.class, name="ENRICHMENT"),
+    @JsonSubTypes.Type(value=ValidationPlugin.class, name="VALIDATION")
 })
 public interface AbstractMetisPlugin {
```

Comment:
Implementing classes are very similar! Maybe we can introduce an abstract class that implements most of the functionality? Also this interface could be renamed, but I am not sure if this is feasible (it is implemented in the metis-core-workflow-qa dependency).

Reply:
Yes indeed I have in mind, just waiting until we have some more plugins ready to see how that looks and probably an abstract class is the way to go.